### PR TITLE
gui: less click+wait, populate timing paths as part of loading timing

### PR DESCRIPTION
**Pull request by oharboe**
This makes the GUI more responsive for large designs when interacting with timing.

report_checks returns instantaneously.

Use GUI_TIMING=0 when timing is not needed.

### CODE REVIEW

File: tools/OpenROAD
Comment (maliberty) [inline tools/OpenROAD:1]: Why is an OR update needed here?

Comment (oharboe) [reply tools/OpenROAD:1]: oops.. fixed.

**Review by maliberty — APPROVED** [review]

**Comment by jeffng-or** [comment]
@oharboe FYI. 

I'm seeing the following issue when trying to pull up the current megaboom:

```
jeffng@boomdev-jeffng: $ inspect_grt/make gui_grt
ODB_FILE=.//results/asap7/BoomTile/base/5_1_grt.odb external/bazel-orfs~~orfs_repositories~docker_orfs/OpenROAD-flow-scripts/tools/install/OpenROAD/bin/openroad -no_init -threads 44  external/bazel-orfs~~orfs_repositories~docker_orfs/OpenROAD-flow-scripts/flow/scripts/open.tcl
Failed to load tclreadline
OpenROAD HEAD-HASH-NOTFOUND 
Features included (+) or not (-): +Charts +GPU +GUI +Python
This program is licensed under the BSD-3 license. See the LICENSE file for details.
Components of this program may be licensed under more restrictive licenses which must be honored.
[INFO ORD-0030] Using 44 thread(s).
GUI_TIMING=1 reading timing, takes a little while for large designs...
estimate_parasitics -global_routing
[WARNING GRT-0097] No global routing found for nets.
[ERROR RSZ-0005] Run global_route before estimating parasitics for global routing.
Error: open.tcl, 59 RSZ-0005
application-specific initialization failed: can't read "auto_index(::tclreadline::ScriptCompleter)": no such element in array
% 
```

Looks pretty related so adding note to this PR. Repro steps:

```
git clone
set up .user-bazelrc
bazel build BoomTile_grt (artifacts exist, so should go quickly)
bazel run BoomTile_grt `pwd`/inspect_grt
inspect_grt/make gui_grt
```

**Comment by oharboe** [comment]
@hovind Any ideas?

`bazel run BoomTile_grt `pwd`/inspect_grt gui_grt` works on my machine, could you try it?

```
$ git log -1
commit 813ad4899babd1f391f81f269dd8db0a0564388a (HEAD, origin/main, origin/HEAD)
Merge: 803437d 04e53e0
Author: Jeff Ng <jeffng@precisioninno.com>
Date:   Mon Oct 14 22:34:01 2024 +0000

    Merge pull request #148 from jeffng-or/move-megaboom-build
    
    Added megaboom build script for CI
$ bazel run BoomTile_grt `pwd`/inspect_grt gui_grt
[deleted]
INFO: Running command line: bazel-bin/BoomTile_grt.sh /home/oyvind/megaboom/inspect_grt gui_grt
ODB_FILE=.//results/asap7/BoomTile/base/5_1_grt.odb external/bazel-orfs~~orfs_repositories~docker_orfs/OpenROAD-flow-scripts/tools/install/OpenROAD/bin/openroad -no_init -threads 16  external/bazel-orfs~~orfs_repositories~docker_orfs/OpenROAD-flow-scripts/flow/scripts/open.tcl
OpenROAD HEAD-HASH-NOTFOUND 
Features included (+) or not (-): +Charts +GPU +GUI +Python
This program is licensed under the BSD-3 license. See the LICENSE file for details.
Components of this program may be licensed under more restrictive licenses which must be honored.
[INFO ORD-0030] Using 16 thread(s).
GUI_TIMING=1 reading timing, takes a little while for large designs...
estimate_parasitics -global_routing
[WARNING GRT-0097] No global routing found for nets.
[ERROR RSZ-0005] Run global_route before estimating parasitics for global routing.
Error: open.tcl, 59 RSZ-0005
openroad> gui::show
```

Note, that I have to run `gui::show` to launch the GUI above is a new and intended behavior(ref @maliberty). I thought we might wanted to restore the previous behavior that the GUI opens even in the event of an error: https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/2456

Your second use-case also works:

```
$ inspect_grt/make gui_grt
ODB_FILE=.//results/asap7/BoomTile/base/5_1_grt.odb external/bazel-orfs~~orfs_repositories~docker_orfs/OpenROAD-flow-scripts/tools/install/OpenROAD/bin/openroad -no_init -threads 16  external/bazel-orfs~~orfs_repositories~docker_orfs/OpenROAD-flow-scripts/flow/scripts/open.tcl
OpenROAD HEAD-HASH-NOTFOUND 
Features included (+) or not (-): +Charts +GPU +GUI +Python
This program is licensed under the BSD-3 license. See the LICENSE file for details.
Components of this program may be licensed under more restrictive licenses which must be honored.
[INFO ORD-0030] Using 16 thread(s).
GUI_TIMING=1 reading timing, takes a little while for large designs...
estimate_parasitics -global_routing
[WARNING GRT-0097] No global routing found for nets.
[ERROR RSZ-0005] Run global_route before estimating parasitics for global routing.
Error: open.tcl, 59 RSZ-0005
openroad> gui::show
```

**Comment by oharboe** [comment]
@jeffng-or I believe that you are seing the problem that https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/2456 "fixes", just run "gui::show" afterwards.

Also, you are seeing an additional problem, but that problem could have been there all along (even if I can't reproduce it), but it might have been "hiding in logging spam" since the GUI opened before.

```
application-specific initialization failed: can't read "auto_index(::tclreadline::ScriptCompleter)": no such element in array
```

**Comment by oharboe** [comment]
Still no luck in reproducing with another use-case:

```
$ . ~/OpenROAD-flow-scripts/env.sh 
OPENROAD: /home/oyvind/OpenROAD-flow-scripts/tools/OpenROAD
$ inspect_grt/make gui_grt
ODB_FILE=.//results/asap7/BoomTile/base/5_1_grt.odb /home/oyvind/OpenROAD-flow-scripts/tools/install/OpenROAD/bin/openroad -no_init -threads 16  /home/oyvind/OpenROAD-flow-scripts/flow/scripts/open.tcl
OpenROAD v2.0-16507-gd0e17f1a2 
Features included (+) or not (-): +Charts +GPU +GUI +Python
This program is licensed under the BSD-3 license. See the LICENSE file for details.
Components of this program may be licensed under more restrictive licenses which must be honored.
[INFO ORD-0030] Using 16 thread(s).
GUI_TIMING=1 reading timing, takes a little while for large designs...
estimate_parasitics -global_routing
[WARNING GRT-0097] No global routing found for nets.
[ERROR RSZ-0005] Run global_route before estimating parasitics for global routing.
Error: open.tcl, 59 RSZ-0005
openroad> 
```

**Comment by oharboe** [comment]
> Failed to load tclreadline

TL;DR the problem is that there is a [hardcoded path](https://github.com/The-OpenROAD-Project/OpenROAD/blob/d0e17f1a2b73a7a72fd004260dcfa44219746b56/src/Main.cc#L376) in OpenROAD to the tcl library. OpenROAD/STA is using tclreadline's example, so this is annoying, but not surprising.

Workarounds:

- use Ubuntu 24.04 (this presumably happens under CentOS?)
- use ". ~/OpenROAD-flow-scripts/env.sh" before running the command aboe
- just type "gui::show" and forego command line completion within OpenROAD GUI for now

**Comment by oharboe** [comment]
> Failed to load tclreadline

@maliberty @jeffng-or @hovind Not pretty, but this would solve it. https://github.com/The-OpenROAD-Project/OpenROAD/pull/5950

I'm loath to try to change OpenSTA or tclreadline...
